js/actionCreators/files.ts

```diff
@@ -198,6 +198,29 @@ export function fetchMediaDuration(url: string, id: number): Dispatchable {
   };
 }
 
+export function loadMedia(
+  e: React.DragEvent<HTMLDivElement>,
+  loadStyle: LoadStyle = LOAD_STYLE.NONE,
+  atIndex = 0
+): Dispatchable {
+  return (dispatch, getState, { handleTrackDropEvent }) => {
+    const defaultHandler = loadFilesFromReferences(
+      e.dataTransfer.files,
+      loadStyle,
+      atIndex
+    );
+
+    if (handleTrackDropEvent)
+      handleTrackDropEvent(e).then(tracks => {
+        if (tracks) dispatch(loadMediaFiles(tracks, loadStyle, atIndex));
+        else {
+          dispatch(defaultHandler);
+        }
+      });
+    else dispatch(defaultHandler);
+  };
```

Comment:
I think you could combine an early return with async/await to make this a bit easier to read:


```JavaScript
if (handleTrackDropEvent)
  const tracks = await handleTrackDropEvent(e);

  if (tracks) {
    return dispatch(loadMediaFiles(tracks, loadStyle, atIndex));
  }
}
return dispatch(defaultHandler);
```

Reply:
Additionally, you could avoid unesesarily calling the default action creator by moving the call to the bottom here:

```JavaScript
if (handleTrackDropEvent)
  const tracks = await handleTrackDropEvent(e);

  if (tracks) {
    dispatch(loadMediaFiles(tracks, loadStyle, atIndex));
    return;
  }
}
dispatch(loadFilesFromReferences(
  e.dataTransfer.files,
  loadStyle,
  atIndex
));